src/streamsync/core_ui.py

```diff
@@ -116,117 +118,244 @@ def to_dict(self) -> Dict:
         active_components = {}
         for id, component in self.components.items():
             active_components[id] = component.to_dict()
+
         return active_components
 
 
-class DependentComponentTree(ComponentTree):
+class ComponentTree():
 
-    def __init__(self, base_component_tree: ComponentTree, attach_root=False):
-        super().__init__(attach_root)
-        self.base_component_tree = base_component_tree
+    def __init__(self, trees: List[ComponentTreeFragment]):
+        assert len(trees) > 0, "Component tree must have at least one tree fragment"
```

Comment:
Not necessarily true for CMC trees, they can be empty if the user chooses to not create any CMCs

Reply:
I renamed trees to tree_branches. A tree can have 0 components but it must still have several branches, one for the bmc, one for the cmc and one for the smc